src/utilities/downloads_backend.jl

```diff
@@ -76,8 +76,9 @@ function _http_request(backend::DownloadsBackend, request::Request, response_str
 
     local buffer
     local response
-    try
-        @repeat 4 try
+
+    get_response =
+        () -> begin
```

Comment:
May as well use the long-form anonymous function syntax
```suggestion
    get_response = function ()
```
Even better option:
```suggestion
    try
        retry(; check, delays) do
```
This eliminates the need to create `get_response_with_retry` and avoid the nasty double call